File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/xmlFactory/GenClass/RiskMonitor-0-1/Excel_valueSideResult.cpp
// Excel_valueSideResult.cpp 
#include "Excel_valueSideResult.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Excel_valueSideResult::Excel_valueSideResult(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //typeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* typeNode = xmlNode->FirstChildElement("type");

   if(typeNode){typeIsNull_ = false;}
   else{typeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- typeNode , address : " << typeNode << std::endl;
   #endif
   if(typeNode)
   {
       type_ = boost::shared_ptr<XsdTypeToken>(new XsdTypeToken(typeNode));
   }

   //currencyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* currencyNode = xmlNode->FirstChildElement("currency");

   if(currencyNode){currencyIsNull_ = false;}
   else{currencyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- currencyNode , address : " << currencyNode << std::endl;
   #endif
   if(currencyNode)
   {
       currency_ = boost::shared_ptr<XsdTypeDouble>(new XsdTypeDouble(currencyNode));
   }

   //dirtyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* dirtyNode = xmlNode->FirstChildElement("dirty");

   if(dirtyNode){dirtyIsNull_ = false;}
   else{dirtyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- dirtyNode , address : " << dirtyNode << std::endl;
   #endif
   if(dirtyNode)
   {
       dirty_ = boost::shared_ptr<XsdTypeDouble>(new XsdTypeDouble(dirtyNode));
   }

   //cleanNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* cleanNode = xmlNode->FirstChildElement("clean");

   if(cleanNode){cleanIsNull_ = false;}
   else{cleanIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- cleanNode , address : " << cleanNode << std::endl;
   #endif
   if(cleanNode)
   {
       clean_ = boost::shared_ptr<XsdTypeDouble>(new XsdTypeDouble(cleanNode));
   }

   //accrualAmtNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* accrualAmtNode = xmlNode->FirstChildElement("accrualAmt");

   if(accrualAmtNode){accrualAmtIsNull_ = false;}
   else{accrualAmtIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accrualAmtNode , address : " << accrualAmtNode << std::endl;
   #endif
   if(accrualAmtNode)
   {
       accrualAmt_ = boost::shared_ptr<XsdTypeDouble>(new XsdTypeDouble(accrualAmtNode));
   }

   //optionNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* optionNode = xmlNode->FirstChildElement("option");

   if(optionNode){optionIsNull_ = false;}
   else{optionIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- optionNode , address : " << optionNode << std::endl;
   #endif
   if(optionNode)
   {
       option_ = boost::shared_ptr<XsdTypeDouble>(new XsdTypeDouble(optionNode));
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeToken> Excel_valueSideResult::getType()
{
   if(!this->typeIsNull_){
        return this->type_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeToken>();
   }
}
boost::shared_ptr<XsdTypeDouble> Excel_valueSideResult::getCurrency()
{
   if(!this->currencyIsNull_){
        return this->currency_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDouble>();
   }
}
boost::shared_ptr<XsdTypeDouble> Excel_valueSideResult::getDirty()
{
   if(!this->dirtyIsNull_){
        return this->dirty_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDouble>();
   }
}
boost::shared_ptr<XsdTypeDouble> Excel_valueSideResult::getClean()
{
   if(!this->cleanIsNull_){
        return this->clean_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDouble>();
   }
}
boost::shared_ptr<XsdTypeDouble> Excel_valueSideResult::getAccrualAmt()
{
   if(!this->accrualAmtIsNull_){
        return this->accrualAmt_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDouble>();
   }
}
boost::shared_ptr<XsdTypeDouble> Excel_valueSideResult::getOption()
{
   if(!this->optionIsNull_){
        return this->option_;
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDouble>();
   }
}
}

File: OTCDerivativesCalculatorModule/Excel_Interface2/GeneratedClass/RiskMonitor-0-1/Excel_valueSideResult.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Excel_valueSideResult : ISerialized
    {
        public Excel_valueSideResult() { }
        public Excel_valueSideResult(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode typeNode = xmlNode.SelectSingleNode("type");
            
            if (typeNode != null)
            {
                if (typeNode.Attributes["href"] != null || typeNode.Attributes["id"] != null) 
                {
                    if (typeNode.Attributes["id"] != null) 
                    {
                        typeIDRef_ = typeNode.Attributes["id"].Value;
                        XsdTypeToken ob = new XsdTypeToken(typeNode);
                        IDManager.SetID(typeIDRef_, ob);
                    }
                    else if (typeNode.Attributes["href"] != null)
                    {
                        typeIDRef_ = typeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        type_ = new XsdTypeToken(typeNode);
                    }
                }
                else
                {
                    type_ = new XsdTypeToken(typeNode);
                }
            }
            
        
            XmlNode currencyNode = xmlNode.SelectSingleNode("currency");
            
            if (currencyNode != null)
            {
                if (currencyNode.Attributes["href"] != null || currencyNode.Attributes["id"] != null) 
                {
                    if (currencyNode.Attributes["id"] != null) 
                    {
                        currencyIDRef_ = currencyNode.Attributes["id"].Value;
                        XsdTypeDouble ob = new XsdTypeDouble(currencyNode);
                        IDManager.SetID(currencyIDRef_, ob);
                    }
                    else if (currencyNode.Attributes["href"] != null)
                    {
                        currencyIDRef_ = currencyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        currency_ = new XsdTypeDouble(currencyNode);
                    }
                }
                else
                {
                    currency_ = new XsdTypeDouble(currencyNode);
                }
            }
            
        
            XmlNode dirtyNode = xmlNode.SelectSingleNode("dirty");
            
            if (dirtyNode != null)
            {
                if (dirtyNode.Attributes["href"] != null || dirtyNode.Attributes["id"] != null) 
                {
                    if (dirtyNode.Attributes["id"] != null) 
                    {
                        dirtyIDRef_ = dirtyNode.Attributes["id"].Value;
                        XsdTypeDouble ob = new XsdTypeDouble(dirtyNode);
                        IDManager.SetID(dirtyIDRef_, ob);
                    }
                    else if (dirtyNode.Attributes["href"] != null)
                    {
                        dirtyIDRef_ = dirtyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        dirty_ = new XsdTypeDouble(dirtyNode);
                    }
                }
                else
                {
                    dirty_ = new XsdTypeDouble(dirtyNode);
                }
            }
            
        
            XmlNode cleanNode = xmlNode.SelectSingleNode("clean");
            
            if (cleanNode != null)
            {
                if (cleanNode.Attributes["href"] != null || cleanNode.Attributes["id"] != null) 
                {
                    if (cleanNode.Attributes["id"] != null) 
                    {
                        cleanIDRef_ = cleanNode.Attributes["id"].Value;
                        XsdTypeDouble ob = new XsdTypeDouble(cleanNode);
                        IDManager.SetID(cleanIDRef_, ob);
                    }
                    else if (cleanNode.Attributes["href"] != null)
                    {
                        cleanIDRef_ = cleanNode.Attributes["href"].Value;
                    }
                    else
                    {
                        clean_ = new XsdTypeDouble(cleanNode);
                    }
                }
                else
                {
                    clean_ = new XsdTypeDouble(cleanNode);
                }
            }
            
        
            XmlNode accrualAmtNode = xmlNode.SelectSingleNode("accrualAmt");
            
            if (accrualAmtNode != null)
            {
                if (accrualAmtNode.Attributes["href"] != null || accrualAmtNode.Attributes["id"] != null) 
                {
                    if (accrualAmtNode.Attributes["id"] != null) 
                    {
                        accrualAmtIDRef_ = accrualAmtNode.Attributes["id"].Value;
                        XsdTypeDouble ob = new XsdTypeDouble(accrualAmtNode);
                        IDManager.SetID(accrualAmtIDRef_, ob);
                    }
                    else if (accrualAmtNode.Attributes["href"] != null)
                    {
                        accrualAmtIDRef_ = accrualAmtNode.Attributes["href"].Value;
                    }
                    else
                    {
                        accrualAmt_ = new XsdTypeDouble(accrualAmtNode);
                    }
                }
                else
                {
                    accrualAmt_ = new XsdTypeDouble(accrualAmtNode);
                }
            }
            
        
            XmlNode optionNode = xmlNode.SelectSingleNode("option");
            
            if (optionNode != null)
            {
                if (optionNode.Attributes["href"] != null || optionNode.Attributes["id"] != null) 
                {
                    if (optionNode.Attributes["id"] != null) 
                    {
                        optionIDRef_ = optionNode.Attributes["id"].Value;
                        XsdTypeDouble ob = new XsdTypeDouble(optionNode);
                        IDManager.SetID(optionIDRef_, ob);
                    }
                    else if (optionNode.Attributes["href"] != null)
                    {
                        optionIDRef_ = optionNode.Attributes["href"].Value;
                    }
                    else
                    {
                        option_ = new XsdTypeDouble(optionNode);
                    }
                }
                else
                {
                    option_ = new XsdTypeDouble(optionNode);
                }
            }
            
        
        }
        
    
        #region Type_
        private XsdTypeToken type_;
        public XsdTypeToken Type_
        {
            get
            {
                if (this.type_ != null)
                {
                    return this.type_; 
                }
                else if (this.typeIDRef_ != null)
                {
                    type_ = IDManager.getID(typeIDRef_) as XsdTypeToken;
                    return this.type_; 
                }
                else
                {
                    throw new Exception( "type_Node no exist!");
                }
            }
            set
            {
                if (this.type_ != value)
                {
                    this.type_ = value;
                }
            }
        }
        #endregion
        
        public string typeIDRef_ { get; set; }
        #region Currency_
        private XsdTypeDouble currency_;
        public XsdTypeDouble Currency_
        {
            get
            {
                if (this.currency_ != null)
                {
                    return this.currency_; 
                }
                else if (this.currencyIDRef_ != null)
                {
                    currency_ = IDManager.getID(currencyIDRef_) as XsdTypeDouble;
                    return this.currency_; 
                }
                else
                {
                    throw new Exception( "currency_Node no exist!");
                }
            }
            set
            {
                if (this.currency_ != value)
                {
                    this.currency_ = value;
                }
            }
        }
        #endregion
        
        public string currencyIDRef_ { get; set; }
        #region Dirty_
        private XsdTypeDouble dirty_;
        public XsdTypeDouble Dirty_
        {
            get
            {
                if (this.dirty_ != null)
                {
                    return this.dirty_; 
                }
                else if (this.dirtyIDRef_ != null)
                {
                    dirty_ = IDManager.getID(dirtyIDRef_) as XsdTypeDouble;
                    return this.dirty_; 
                }
                else
                {
                    throw new Exception( "dirty_Node no exist!");
                }
            }
            set
            {
                if (this.dirty_ != value)
                {
                    this.dirty_ = value;
                }
            }
        }
        #endregion
        
        public string dirtyIDRef_ { get; set; }
        #region Clean_
        private XsdTypeDouble clean_;
        public XsdTypeDouble Clean_
        {
            get
            {
                if (this.clean_ != null)
                {
                    return this.clean_; 
                }
                else if (this.cleanIDRef_ != null)
                {
                    clean_ = IDManager.getID(cleanIDRef_) as XsdTypeDouble;
                    return this.clean_; 
                }
                else
                {
                    throw new Exception( "clean_Node no exist!");
                }
            }
            set
            {
                if (this.clean_ != value)
                {
                    this.clean_ = value;
                }
            }
        }
        #endregion
        
        public string cleanIDRef_ { get; set; }
        #region AccrualAmt_
        private XsdTypeDouble accrualAmt_;
        public XsdTypeDouble AccrualAmt_
        {
            get
            {
                if (this.accrualAmt_ != null)
                {
                    return this.accrualAmt_; 
                }
                else if (this.accrualAmtIDRef_ != null)
                {
                    accrualAmt_ = IDManager.getID(accrualAmtIDRef_) as XsdTypeDouble;
                    return this.accrualAmt_; 
                }
                else
                {
                    throw new Exception( "accrualAmt_Node no exist!");
                }
            }
            set
            {
                if (this.accrualAmt_ != value)
                {
                    this.accrualAmt_ = value;
                }
            }
        }
        #endregion
        
        public string accrualAmtIDRef_ { get; set; }
        #region Option_
        private XsdTypeDouble option_;
        public XsdTypeDouble Option_
        {
            get
            {
                if (this.option_ != null)
                {
                    return this.option_; 
                }
                else if (this.optionIDRef_ != null)
                {
                    option_ = IDManager.getID(optionIDRef_) as XsdTypeDouble;
                    return this.option_; 
                }
                else
                {
                    throw new Exception( "option_Node no exist!");
                }
            }
            set
            {
                if (this.option_ != value)
                {
                    this.option_ = value;
                }
            }
        }
        #endregion
        
        public string optionIDRef_ { get; set; }
        
    
        
    
    }
    
}

